src/bin/lib/render-plan.ts

```diff
@@ -0,0 +1,24 @@
+import chalk from 'chalk'
+import { renderRuntimeErrorsAndWriteToFile, renderValidationErrors } from './plan-messages'
+import renderMessages from './plan-messages'
+import { RequestBatch } from '../../lib/offline-api/index'
+
+const renderPlan = async (batches: RequestBatch[], errorsFile: string): Promise<void> => {
+  const hasValidationErrors = batches.some(batch => batch.validationErrors.length > 0)
+  const hasRuntimeErrors = batches.some(batch => batch.runtimeErrors.length > 0)
+
+  if (hasValidationErrors || hasRuntimeErrors) {
+    if (hasValidationErrors) {
+      renderValidationErrors(batches)
+    } else {
+      await renderRuntimeErrorsAndWriteToFile(batches, errorsFile)
+    }
+    console.log(chalk`🚨  {bold.red Migration unsuccessful}`)
+    process.exit(1)
+  }
+
+  console.log(chalk`{bold.green The following migration has been planned}\n`)
+  renderMessages(batches)
+}
+
+export default renderPlan
```

Comment:
I think this function is doing too many things at once:

- It'll render the plan
- (potentially) it'll write the errors to a file
- (potentially) it'll exit the process

What do you think if we pull the error handling code out of `renderPlan` so we endup with something like this:

(in `src/bin/cli.ts`)

```javascript
const errorsInPlan = batches.getErrors() // or something else

if (errorsInPlan) {
   await renderErrors(errorsInPlan) // this does the "if (hasValidationErrors || hasRuntimeErrors) {" logic

  process.exit(1)
}

await renderPlannedMigration(batches, errorsFile)

```